arangod/Cluster/ClusterMethods.cpp

```diff
@@ -3750,10 +3750,17 @@ arangodb::Result hotBackupCoordinator(ClusterFeature& feature, VPackSlice const
       return result;
     }
 
+    auto releaseAgencyLock = scopeGuard([&]{
+      LOG_TOPIC("52416", INFO, Logger::BACKUP)
+          << "Releasing agency lock with scope guard! backupId: " << backupId;
+      ci.agencyHotBackupUnlock(backupId, timeout, supervisionOff);
+    });
+
     if (end < steady_clock::now()) {
       LOG_TOPIC("352d6", INFO, Logger::BACKUP)
           << "hot backup didn't get to locking phase within " << timeout << "s.";
       auto hlRes = ci.agencyHotBackupUnlock(backupId, timeout, supervisionOff);
```

Comment:
I notice that `hlRes` is ignored here, and generally the return value of the calls to `agencyHotBackupUnlock` is ignored everywhere. Plus `agencyHotBackupUnlock` does not even log anything on error. So if for any reason whatsoever the request fails, we don't get a hint that it happened, nor why.

If you're already on this, maybe its worthwhile adding some logs, and maybe add some error handling in case the unlock fails - however that should look like. Maybe it has to be retried until we know the lock is gone? I really don't know anything about the process, so maybe it has to be something different, but it might be worth thinking about.